src/app/clusters/pump-configuration-and-control-server/pump-configuration-and-control-server.cpp

```diff
@@ -33,7 +36,24 @@ void MatterPumpConfigurationAndControlClusterServerAttributeChangedCallback(cons
 {
     emberAfDebugPrintln("PCC Server Cluster Attribute changed [EP:%d, ID:0x%x]", attributePath.mEndpointId,
                         (unsigned int) attributePath.mAttributeId);
-    // TODO
+
+    switch (attributePath.mAttributeId)
+    {
+    case ControlMode::Id: {
+        uint8_t value;
+        ControlMode::Get(attributePath.mEndpointId, &value);
+        EffectiveControlMode::Set(attributePath.mEndpointId, value);
```

Comment:
I don't understand how the application is expected to implement the spec requirements here.  The spec says:

> It will have the value of the ControlMode attribute, unless a remote sensor is used as the sensor for regulation of the pump. In this case, EffectiveControlMode will display ConstantPressure, ConstantFlow or ConstantTemperature if the remote sensor is a pressure sensor, a flow sensor or a temperature sensor respectively, regardless of the value of the ControlMode attribute.

This code runs after the application-level attribute change callback.  So how is an application expected to maintain the proper value of the effective control mode in the "remote sensor" case?

Reply:
Adding sensors is optional features in PCC. The optional feature to add sensors for PCC is not used by any DUT's prepared for Matter 1.0. I was under the impression that we are not required to implement optional features when they are not used by any DUT's for Matter 1.0?

Do we have a good way to mark the optional(ControlMode: ConstantPressure, ConstantFlow and ConstantTemperature) PCC features as not implemented for Matter 1.0?  We can also add the optional sensor functionality to the example project, but it requires additional work.

Reply:
Just saying that we don't support that case in 1.0 is fine, if that's the intent.  I just wanted to make sure I understood how things fit together.

It's worth adding a comment to this code that explains that we don't support that right now and if we do add that support this code will need to be revisited.

Reply:
@bzbarsky-apple OK, we can add the structure for Controlmodes: ConstrantPressure, ProportionalPressure, ConstantFlow, ConstantTemperature and then write not implemented for 1.0.

Looking further into it [ ControlMode Attribute](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/PumpConfigurationControl.adoc#ref_sec_6_2_2_2_3_1)..... Each vendor implementation will support different sets of ControlModes.  Rater than just returning an error when trying to set a ControlMode when it is not implemented  it would probably benefit if we have the Controlmodes as a Featuremap: ConstantSpeed, ConstantPressure, ProportionalPressure, ConstantFlow, ConstantTemperature and Automatic, right? 

Then it would be straight forward for the controller to see which ControlMode elements are supported for a specific vendor implementation. Typical selection of ControlModes would be to always to support ConstantSpeed (0) and Automatic (7) and then one or two of the other ControlModes: ConstrantPressure, ProportionalPressure, ConstantFlow, ConstantTemperature.

Hmmm, thinking it through again.... The reason why we didn't do the featuremap in the first place was because of the resistance and the strings back to the "OLD ZCL" implementation this gives us some limitation on how much we can change here.

Reply:
Right, I am not asking for spec changes here, by any means.  And I am not even asking for impl changes now that I understand the intent.  We should just document the limitations of this code...

Reply:
OK, understood. We can revisit this post V1.0 when this part is required. Thanks for input.